kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala

```diff
@@ -53,4 +57,72 @@ private[v1] class AdminResource extends ApiRequestContext with Logging {
     KyuubiServer.reloadHadoopConf()
     Response.ok(s"Refresh the hadoop conf for ${fe.connectionUrl} successfully.").build()
   }
+
+  @ApiResponse(
+    responseCode = "200",
+    content = Array(new Content(
+      mediaType = MediaType.APPLICATION_JSON)),
+    description = "delete kyuubi engine")
+  @DELETE
+  @Path("engine")
+  def deleteEngine(
+      @QueryParam("user") user: String,
+      @QueryParam("type") engineType: String,
+      @QueryParam("subdomain") engineSubdomain: String,
+      @QueryParam("sharelevel") engineShareLevel: String,
+      @QueryParam("version") version: String,
+      @QueryParam("namespace") namespace: String): Response = {
+    val userName = fe.getUserName(Map.empty)
+    val ipAddress = fe.getIpAddress
+    info(s"Received delete engine request from $userName/$ipAddress")
+    if (!userName.equals(administrator)) {
```

Comment:
this api is open for any user.

You can refer https://github.com/apache/incubator-kyuubi/blob/0c2091cd03c92f4b3e402a2dc84ef35cdc5d5bea/kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala#L308

to get the userName